wrappers/src/app_cs.cpp

```diff
@@ -116,19 +121,19 @@ extern "C" {
     REALM_EXPORT void shared_app_initialize(uint16_t* platform, size_t platform_len,
         uint16_t* platform_version, size_t platform_version_len,
         uint16_t* sdk_version, size_t sdk_version_len,
-        decltype(s_user_callback) user_callback,
-        decltype(s_void_callback) void_callback,
-        decltype(s_bson_callback) bson_callback,
-        decltype(s_log_message_callback) log_message_callback)
+        UserCallbackT user_callback,
+        VoidCallbackT void_callback,
+        BsonCallbackT bson_callback,
+        LogMessageCallbackT log_message_callback)
     {
         s_platform = Utf16StringAccessor(platform, platform_len);
         s_platform_version = Utf16StringAccessor(platform_version, platform_version_len);
         s_sdk_version = Utf16StringAccessor(sdk_version, sdk_version_len);
 
-        s_user_callback = user_callback;
-        s_void_callback = void_callback;
-        s_bson_callback = bson_callback;
-        s_log_message_callback = log_message_callback;
+        s_user_callback = wrap_managed_callback(user_callback);
```

Comment:
And I assume this and the ones below are `std::function`s because of the issue you describe in `marshalling.hpp`, right?

Reply:
Yep - I don't believe it's possible to avoid changing the type.